### PR TITLE
MQ Ice Cavern Rework

### PR DESCRIPTION
**Pull request by Joshua8600**

### CODE REVIEW

File: packages/data/src/world/mq/ice_cavern_mq.yml [hunk 1]
```diff
@@ -2,42 +2,98 @@
...
+  events:
+    ICE_MQ_FIRST_CRYSTAL: "has_ranged_weapon || has_explosives || (is_adult && (has(SWORD_BIGGORON) || can_use_mask_blast))" #Biggoron Jump Slash as Adult hits it, so does Adult Blast Mask. The pot carry is very trick territory; this crystal switch is a problem.
+"Ice Cavern After Initial Ice Block":
+  dungeon: IC
+  exits:
+    "Ice Cavern Room After Entrance Boulder": "event(ICE_MQ_FIRST_CRYSTAL)"
+    "Ice Cavern Main Room": "true"
   locations:
     "MQ Ice Cavern Pot First Room 1": "true"
     "MQ Ice Cavern Pot First Room 2": "true"
```
Comment (XenoWars) [inline packages/data/src/world/mq/ice_cavern_mq.yml:23]: This doesn't seem right. aren't these pots in the first corridor, which is why you are even able to throw a pot at the crystal?

File: packages/data/src/world/mq/ice_cavern_mq.yml [hunk 2]
```diff
@@ -2,42 +2,98 @@
...
+    "MQ Ice Cavern Piece of Heart": "has_explosives || (can_use_mask_blast && (is_child || (is_adult && has_shield)))" #Blast Mask can only hit when adult if crouching with shield.
+    "MQ Ice Cavern GS Compass Room": "gs && ((can_play_time && has_blue_fire && can_damage_skull) || has_blue_fire_arrows || can_boomerang)"
     "MQ Ice Cavern Pot Compass Room 1": "true"
     "MQ Ice Cavern Pot Compass Room 2": "true"
-"Ice Cavern Big Room":
+"Ice Cavern Vanilla Push Block Room":
+  dungeon: IC
+  exits:
+    "Ice Cavern Main Room": "has_blue_fire || hookshot_anywhere"
+    "Ice Cavern Vanilla Push Block Room Scarecrow Ledge": "climb_anywhere || hookshot_anywhere || scarecrow_hookshot ||(is_adult && trick(OOT_MQ_ICE_SCARE_NOTHING)) || (has_hover_boots && (can_longshot || (is_adult && trick(OOT_MQ_ICE_SCARE_HOVER))))"
```
Comment (XenoWars) [inline packages/data/src/world/mq/ice_cavern_mq.yml:64]: Spacing at the first parenthesis

File: packages/data/src/world/mq/ice_cavern_mq.yml [hunk 3]
```diff
@@ -2,42 +2,98 @@
...
+    "Ice Cavern Past Red Ice Before Shiek": "climb_anywhere || hookshot_anywhere || (has_blue_fire && (is_adult || (can_play_time && has_hover_boots)))"
```
Comment (XenoWars) [inline packages/data/src/world/mq/ice_cavern_mq.yml:65]: Couldn't this just be Before Shiek if there isn't one for not being past the red ice?

**Review by XenoWars — CHANGES_REQUESTED** [review]

File: packages/data/src/world/mq/ice_cavern_mq.yml (hunk 1 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/ice_cavern_mq.yml:23]: No. The first corridor has one pot: "MQ Ice Cavern Pot Entrance"

These pots Nax labelled semi-incorrectly. THey are right after the ice block where you need to hit the crystal.

File: packages/data/src/world/mq/ice_cavern_mq.yml (hunk 3 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/ice_cavern_mq.yml:65]: This is in prep for Door Rando.

Passed Red Ice Before Shielk > Serenade Entry Door > Serenade Exit Door > Serenade Room

File: packages/data/src/world/mq/ice_cavern_mq.yml (hunk 2 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/ice_cavern_mq.yml:64]: Fixed.

File: packages/data/src/world/mq/ice_cavern_mq.yml (hunk 3 above)
Comment (XenoWars) [reply packages/data/src/world/mq/ice_cavern_mq.yml:65]: After some thinking, I recognize specifying being past the red ice is necessary, due to vanilla Ice Cavern

**Review by XenoWars — APPROVED** [review]

**Pull request merged**